### ISSUE TITLE
Endpoint classes during dragging

### ISSUE THREAD

**Issue by panayotisk** [jsplumb/jsplumb#1124]
**version:** jsplumb 5.5.5
**description:** While dragging a source endpoint the css classes on the endpoint element(s) are different if the connection is created with the mouse from a connection created programmatically. The source endpoint that is being dragged.
1. After a call to connect() in the code sandbox given we get the following classes on the endpoint elements in the developer tools: 
![image](https://user-images.githubusercontent.com/44521424/164912753-3374139a-7bb0-4fbf-97e5-3cf7e76b5cc7.png)
Detaching the source endpoint and dragging:
![image](https://user-images.githubusercontent.com/44521424/164912885-28eb04de-ce2f-4d4f-bfe0-625d41abc1d9.png)
We get the following:
![image](https://user-images.githubusercontent.com/44521424/164912927-5f4331f0-4197-44c4-8fa1-1ee788d86959.png)
and:
![image](https://user-images.githubusercontent.com/44521424/164913019-31a8b432-0af1-482c-a479-9a39f391f957.png)
So the classes jtk-endpoint-connected, jtk-endpoint-full and jtk-dragging are applied to the original endpoint (the one still on the element), while the temporary dragged endpoint has the class jtk-endpoint.
2. If we connect the elements using the mouse the two endpoints change roles regarding their classes. At first we see:
![image](https://user-images.githubusercontent.com/44521424/164913847-56680fd4-661b-4bae-b1ec-3ef4fa143e30.png)
And while dragging the source endpoint we get:
![image](https://user-images.githubusercontent.com/44521424/164913914-562d94f4-1ca2-4d6b-b437-02adf62f010e.png)
So this time the jtk-endpoint-connected, jtk-endpoint-full and jtk-dragging are applied to the temporary dragged endpoint.
**expected behavior:** One would expect to have a consistent behavior on source and target endpoints (I, personally, would lean towards the behavior of the 2nd example).
**notes:** Consistent appliance of jsplumb classes is quite important if we need to style endpoints via css. 
Dragging a target endpoint applies the above classes to the original endpoint in both cases (connect() or mouse)!

**Comment by panayotisk** [comment]
A code sandbox for playground: [https://codesandbox.io/s/messed-up-classes-hp259s?file=/src/index.js](url)

**Comment by sporritt** [comment]
I agree about consistency.  I'm not sure about the specific though..I have conflicting opinions. 

I think the temporary endpoint (what is referred to as a "floating" endpoint in the source) should probably have a distinct class applied to it - `jtk-floating-endpoint`, for instance. This would make it unambiguous what that endpoint was doing.

By dint of the fact that a floating endpoint only supports one connection, it is, by default, always full, and it is also always connected, since it wouldn't exist otherwise. So it could be argued that neither `jtk-endpoint-connected` nor `jtk-endpoint-full` are required on the floating endpoint.

On the other hand, if on the stationary endpoint those classes have been used to set some visual cues, if the desire is to maintain those visual cues whenever an existing connection is detached, the classes would need to be copied over to the floating endpoint.  But I don't think that I have ever seen someone want to do this.

Another question is what should happen to the stationary endpoint when an existing connection is dragged off of it, ie. if it has no other connections then it is no longer "full" or "connected", and theoretically the related classes should be removed.

As for `jtk-dragging`, the original intention of that class was to allow a user to indicate when a new connection was being dragged from an endpoint. The fact that it is applied when a source connection is dragged off is a quirk of the code, and ideally would not happen.

So, in conclusion, I propose to set it up like this:

- `jtk-endpoint-connected` and `jtk-endpoint-full` are only ever applied to stationary endpoints, and may be unapplied when an existing connection is dragged off of a stationary endpoint (either source or target)
- floating endpoints will have a `jtk-floating-endpoint` class assigned
- `jtk-dragging` will only ever be assigned to a stationary source endpoint during the drag of a new connection, and in no other circumstance
- both stationary and floating endpoints will have a `jtk-endpoint` class

**Comment by panayotisk** [comment]
Creating a new distinct class for floating endpoints (`jtk-floating-endpoint`) would surely help distinguishing them in css. I also agree that there should be no need for `jtk-endpoint-connected` or `jtk-endpoint-full` since the above class would imply them.
In conclusion I agree with all the proposals. In my case probably I would not be using `jtk-dragging` at all.

I do need the existing custom classes of the stationary endpoint on the floating endpoint in my specific use-case. I style the color of the endpoint by a custom class such as `theme-red`, `theme-blue` etc. I haven't managed to find a cleaner way to colorize the connections and their endpoints other than giving all of them a specific class for color (this is why the new cascade property in [addClass ](https://docs.jsplumbtoolkit.com/community/apidocs/core.component.addclass) has proven very handy).  So, I could use these classes on the floating endpoint to colorize it also. Or maybe a way to add all (or some) of them myself somehow upon the creation of the floating endpoint.

**Comment by sporritt** [comment]
Yes, on this last point of yours: the floating endpoint should be a clone of the endpoint it originated from, that is true, and user-added CSS classes should therefore be copied over when creating the floating endpoint.

**Comment by sporritt** [comment]
5.6.0 was released yesterday with these updates.

**Comment by panayotisk** [comment]
I will be testing the updates in the next few days. Thank you for fixing the issues!

**Comment by panayotisk** [comment]
One little thing I noticed is it did not copy the css classes defined on the endpoint (EndpointSpec.options), while the dynamically added classes (addClass call) are copied. I can work around that, just mentioning it in case you believe this behavior should be fixed. Thank you for the update!
